lib/cli.js

```diff
@@ -127,17 +128,27 @@ function buildWatcherOptions(options) {
   if (!options) {
     options = {};
   }
-  const watcher = options.watcher;
+  let watcher = options.watcher;
 
   if (watcher && ['polling', 'watchman', 'node', 'events'].indexOf(watcher) === -1) {
     throw new Error(`Unknown watcher type --watcher=[polling|watchman|node] but was: ${watcher}`);
   }
 
+  if (!watcher) {
+    try {
+      childProcess.execSync('watchman version');
+      // if `watchman version` did not throw, the watchman is installed and can be used
+      watcher = 'watchman';
```

Comment:
A little insider knowledge, I believe we should only support watchman `>= 3.7` as it has a feature which automatically shares common watches if possible.

This version has also been around for quite some time, so it should be perfectly reasonable for us to do that.

```
let version = execSync('watchman version');
semver.satisfies(version, '>= 3.7.0'),
```

---

For reference, ember-cli has the following: https://github.com/ember-cli/ember-cli/blob/master/lib/models/watch-detector.js#L197 Which tests many failure scenarios and tries to choose the best possible watcher.

I would not be apposed to us as a next  step, extracting that into a common module which can be used in both projects. As it has been pretty thoroughly battle tested, and comes with some handy tests: https://github.com/ember-cli/ember-cli/blob/master/tests/unit/models/watch-detector-test.js

If extracted, I could put it up at `ember-cli/watch-detector`, thoughts?

Reply:
makes complete sense. lets do that. I will prepare the project and when ready we can move it under ember-cli organization

Reply:
:+1: